src/createServer.js

```diff
@@ -1,3 +1,61 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+
+    res.setHeader('Content-Type', 'application/json');
+
+    const originalText = normalizedURL.pathname.slice(1);
+    const targetCase = Object.fromEntries(
+      normalizedURL.searchParams.entries(),
+    ).toCase;
+
+    const errors = [];
+    const noTextToConvert = 'Text to convert is required. '
+    + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".';
+    const noToCase = '"toCase" query param is required. '
+    + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".';
+    const invalidCase = 'This case is not supported. '
+    + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.';
+
+    const caseList = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    if (!originalText) {
+      errors.push({ message: noTextToConvert });
+    };
+
+    if (!targetCase) {
+      errors.push({ message: noToCase });
+    } else if (!(caseList.find(el => el === targetCase))) {
+      errors.push({ message: invalidCase });
+    }
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.statusText = 'Bad request';
+
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const { originalCase, convertedText } = convertToCase(
+      originalText, targetCase,
+    );
+
+    res.statusCode = 200;
+
+    res.end(JSON.stringify({
+      originalCase,
+      targetCase,
+      originalText,
+      convertedText,
+    }));
+  });
```

Comment:
You can put it in a condition with errors.length in the else block and then you don't have to write return